### PR TITLE
make route based metrics optional

### PR DESCRIPTION
**Pull request by aryszka**
with large number of routes (~400k), the number of collected metrics can get too high. This PR provides options to control the collection of metrics whose keys vary by route ID.

These metrics were enabled by default, to keep this behavior backwards compatible, this PR introduces an additional option: DisableCompatibilityDefaults. Without settings this flag, the new metrics options take no effect.

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -182,12 +226,16 @@ func (m *Metrics) MeasureFilterResponse(filterName string, start time.Time) {
 }
 
 func (m *Metrics) MeasureAllFiltersResponse(routeId string, start time.Time) {
-	m.measureSince(fmt.Sprintf(KeyFiltersResponse, routeId), start)
+	if !m.options.EnableAllFiltersMetrics {
```
Comment (vigneshshanmugam) [inline metrics/metrics.go]: shouldn't this be other way around?

Comment (aryszka) [reply metrics/metrics.go]: oh.. yeah. pushing, thx!

Comment (vigneshshanmugam) [reply metrics/metrics.go]: Cool

**Comment by vigneshshanmugam** [comment]
👍

**Comment by vigneshshanmugam** [comment]
👍

**Comment by aryszka** [comment]
:+1:

**Comment by NorthFury** [comment]
I don't really like `DisableCompatibilityDefaults` but I can't see a better way of handling this.

**Comment by NorthFury** [comment]
:+1:

**Comment by aryszka** [comment]
missed one, please recheck :(

**Comment by NorthFury** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**